NSrtm.Core/Pgm/Grid/PgmGridConstants.cs

```diff
@@ -0,0 +1,125 @@
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace NSrtm.Core.Pgm.Grid
+{
+    public struct PgmGridConstants
+    {
+        private readonly double _offset;
+        private readonly double _scale;
+        private readonly int _orginLat;
+        private readonly int _orginLon;
+        private readonly int _gridWidthPoints;
+        private readonly int _gridHeightPoints;
+        private readonly int _maxValue;
+        private readonly int _preambleLength;
+        private readonly int _numberOfPoints;
+
+        public PgmGridConstants(
+            double offset,
+            double scale,
+            int orginLat,
+            int orginLon,
+            int gridWidthPoints,
+            int gridHeightPoints,
+            int maxValue,
+            int preambleLength)
+        {
+            _offset = offset;
+            _scale = scale;
+            _orginLat = orginLat;
+            _orginLon = orginLon;
+            _gridHeightPoints = gridHeightPoints;
+            _gridWidthPoints = gridWidthPoints;
+            _maxValue = maxValue;
+            _preambleLength = preambleLength;
+            _numberOfPoints = _gridHeightPoints * _gridWidthPoints;
+        }
+
+        public double Offset { get { return _offset; } }
+        public double Scale { get { return _scale; } }
+        public int OrginLat { get { return _orginLat; } }
+        public int OrginLon { get { return _orginLon; } }
+        public int GridWidthPoints { get { return _gridWidthPoints; } }
+        public int GridHeightPoints { get { return _gridHeightPoints; } }
+        public int MaxValue { get { return _maxValue; } }
+        public int NumberOfPoints { get { return _numberOfPoints; } }
+        public int PreambleLength { get { return _preambleLength; } }
+    }
+
+    internal static class PgmGridConstantsExtractor
+    {
+        private static readonly Regex LatLonParser = new Regex("^(?<deg>[-+0-9]+)+(?<pos>[ENSW]*)$");
+
+        public static PgmGridConstants FromStream(Stream stream)
+        {
+            var preamble = extractPreambleFromStream(stream);
+            return getConstatantsFromPreamble(preamble);
+        }
+
+        private static string extractPreambleFromStream(Stream stream)
+        {
+            using (var reader = new StreamReader(stream, Encoding.UTF8, false, 1024, true))
+            {
+                const string _pgmMagicNumber = "P5";
+                var preamble = reader.ReadLine();
+                if (preamble != _pgmMagicNumber)
+                {
+                    throw new FileFormatException(String.Format("Wrong pgm magic number. Actual {0}, but should be {1}.", preamble, _pgmMagicNumber));
+                }
+
+                string lastLine;
+                do
+                {
+                    lastLine = reader.ReadLine();
+                    preamble += lastLine;
```

Comment:
IMO because this method is less sensitive to changes in the file. I use magic numbers only for data whose place is defined by pgm specifications. Of course i could use huge regex but it would take me a while to create it.